upgraderesponder/service.go

```diff
@@ -49,12 +46,21 @@ var (
 )
 
 type Server struct {
```

Comment:
I was trying to avoid making more changes than necessary to the original code to facilitate pulling changes from upstream in the future. I've left this for now, but I'm happy to change it if you insist.